drivers/timer/stm32_lptim_timer.c

```diff
@@ -429,6 +429,35 @@ static int sys_clock_driver_init(void)
 	}
 #endif
 
+#if DT_PROP(DT_DRV_INST(0), st_counter_value)
+	/* Calculate the corresponding prescaler in range [1..128] */
+	uint8_t presci;
+
+	for (presci = 0; presci <= 7 ; presci++) {
```

Comment:
Now that `st_counter_value` could only be with bounds defined by `st_prescaler`, can't we use a faster check ?
For instance  if st_counter_value > (lptim_clock_freq / lptim_clock_presc) * 0xFFFF